From 657828bf677e2c49e45aed0da180665c2c238232 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Wed, 9 Oct 2024 12:08:06 +0200
Subject: [PATCH 7/7] gtkwaylandsink: config buffer pool with query size when
 propose_allocation

Backport of waylandsink: config buffer pool with query size when propose_allocation
on gtkwaylandsink (was missing upstream).

Upstream-Status: Submitted
---
 ext/gtk/gstgtkwaylandsink.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ext/gtk/gstgtkwaylandsink.c b/ext/gtk/gstgtkwaylandsink.c
index c198954..fbe05be 100644
--- a/ext/gtk/gstgtkwaylandsink.c
+++ b/ext/gtk/gstgtkwaylandsink.c
@@ -1047,6 +1047,13 @@ gst_gtk_wayland_sink_propose_allocation (GstBaseSink * bsink, GstQuery * query)
 
   gst_query_parse_allocation (query, &caps, &need_pool);
 
+  if (caps == NULL)
+    return FALSE;
+
+  /* extract info from caps */
+  if (!gst_video_info_from_caps (&priv->video_info, caps))
+    return FALSE;
+
   if (need_pool) {
     GstStructure *config;
     pool = gst_wl_video_buffer_pool_new ();
-- 
2.25.1

